codes/ruby/chapter_stack_and_queue/deque.rb

```diff
@@ -0,0 +1,41 @@
+=begin
+File: deque.rb
+Created Time: 2024-04-06
+Author: Xuan Khoa Tu Nguyen (ngxktuzkai2000@gmail.com)
+=end
+
+### Driver Code ###
+
+# 初始化双向队列
+# Ruby 没有内直的双端队列，只能把 Array 当作双端队列来使用
+deque = []
+
+# 元素如队
+deque << 2
+deque << 5
+deque << 4
+# 请注意，由于是数组，Array#unshift 方法的时间复杂度为 O(n)
+deque.unshift(3)
```

Comment:
This method, as in this [comment](https://stackoverflow.com/a/47683752) stated, has been armotized to be `O(1)` in average, but an individual operation may be `O(n)`. So I keep the comment to be `O(n)` statement.

Reply:
That's fine. The worst time complexity is `O(n)`